File: gcd.c
int gcd(int a, int b) {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
}

void main() {
    int a = 24;
    int b = 18;
    int result = gcd(a, b);
}

File: gcd.rs
fn gcd(a: u64, b: u64) -> u64 {
    if b == 0 {
        return a;
    }
    gcd(b, a % b)
}

fn main() {
    let a = 24;
    let b = 18;
    let result = gcd(a, b);
}
